docs/design/proposals/labels.md

```diff
@@ -0,0 +1,109 @@
+
+Specifically, we need a clearer way to identify:
+1.  Which RGD is responsible for reconciling a given instance.
+2.  Which RGD a created resource originates from.
+3.  Which instance a created resource belongs to.
+
+This proposal aims to introduce new, more descriptive labels to clarify these relationships, aligning with Kubernetes best practices for resource management and observability.
+
+## Proposal
+
+We propose introducing a set of standardized labels to better represent the relationships between RGDs, instances, and the resources they manage. This will improve the ability to discover, manage, and debug resources created and managed by KRO.
+
+#### Overview
+
+The proposal is to add the following labels:
+-   A label on an instance to indicate which RGD is reconciling it.
+-   A label on resources created by an instance reconciliation to indicate which RGD they come from.
+-   Labels on resources created by an instance reconciliation to indicate which instance created them.
+
+This aligns with the common Kubernetes practice of using labels for discovery and organization of related resources, as seen in tools like Helm and Kustomize.
+
+#### Design details
+
+We will introduce the following new labels, which will be defined in `@pkg/metadata/labels.go`:
+
+1.  **Label for an instance being reconciled:**
+    - **Existing Labels:**
+      - kro.run/resource-graph-definition-id: 761e8fb7-14d5-4da1-b4f9-0a16fc334d6c
+      - kro.run/resource-graph-definition-name: check-instance-creation
+      - Conflicts with label used in creation path for nested RGD (https://github.com/kro-run/kro/pull/631)
+    - **Proposed Label:**
+      - `kro.run/reconciled-by-rgd`
+    - **Value:** The name of the ResourceGraphDefinition (e.g., `my-rgd-name`).
+    - **Purpose:** To indicate that an instance is being actively reconciled against a specific RGD. This label will be applied to the instance itself.
+    - **Alternative Names:**
+      - `kro.run/reconciling-rgd`
+      - `kro.run/template`
+
+2.  **Label for resources created during instance reconciliation:**
+    - **Existing Labels:**
+      - kro.run/resource-graph-definition-id: 761e8fb7-14d5-4da1-b4f9-0a16fc334d6c
+      - kro.run/resource-graph-definition-name: check-instance-creation
+    - **Proposed Label:**
+      - `kro.run/defined-by-rgd`
+    - **Value:** The name of the ResourceGraphDefinition (e.g., `my-rgd-name`).
+    - **Purpose:** To identify which RGD was used as a template to create a resource during the reconciliation of an instance. This label will be applied to all resources created by the instance controller.
+    - **Alternative Names:**
+      - kro.run/created-by-rgd
+      - kro.run/managed-by-rgd
```

Comment:
I think managed-by probably fits best since we continuously work on it

Reply:
I think if we have `kro.run/managed-by-rgd` it could be reused for the instance too. Is there a specific need to have two labels here?